### PR TITLE
Added support to source the grabber traffic from a specific IP address / interface on the machine.

### PR DESCRIPTION
**Pull request by timapril**
The IP in question is checked to make sure that it is configured
for the device to reduce the ability to spoof.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -140,6 +145,36 @@ func init() {
 		zlog.Fatalf("--connections-per-host must be in the range [0,50]")
 	}
 
+	// If a source IP is specified, verify that the host is advertising the IP
+	// and has a chance to receive the responses.
+	if sourceIP != "" {
+		config.LocalAddressSet = false
+		providedAddr := net.ParseIP(sourceIP)
```
Comment (dadrian) [inline main.go]: On error, this can return nil. We should `Fatalf` if that is the case.

File: main.go [hunk 2]
```diff
@@ -140,6 +145,36 @@ func init() {
...
+		for idx, i := range localInterfaces {
+			addresses, adddressesErr := i.Addrs()
+			if adddressesErr != nil {
+				zlog.Fatalf("Error getting Local addresses for interface %d: %s", idx, adddressesErr.Error())
+			}
+			for _, address := range addresses {
+				castAddress := address.(*net.IPNet)
+				if castAddress.Contains(providedAddr) {
+					tcpAddr := &net.TCPAddr{
+						IP: address.(*net.IPNet).IP,
```
Comment (dadrian) [inline main.go]: This can be `IP: castAddress.IP`

File: main.go (hunk 1 above)
Comment (timapril) [inline main.go]: Good catch, I just noticed that my errcheck addon was turned off. I will update that shortly.

File: main.go (hunk 2 above)
Comment (timapril) [inline main.go]: This appears to work. I thought I tried that and got an odd error. I will roll that into the new version.

**Comment by timapril** [comment]
The machine that I have been testing on has 2 interfaces, one that has 2 v4 addresses and a v6 address and then another interface that is not enabled. With this change, I am able to send all of the traffic out of one of the virtual interfaces on one NIC.

**Comment by zakird** [comment]
I'm hesitant to split copyright for this project as proposed. We have yet so far managed to avoid this and longterm, I'd prefer to not set a precedent where the copyright for small chunks of code are held by different groups. Does Akamai require this, or can copyright be assigned to Michigan?

**Comment by dadrian** [comment]
I'm closing this PR until the copyright issue is resolved.

**Comment by timapril** [comment]
We have submitted a CLA so the copyright issue should be addressed now. @zakird will probably have more information.

**Pull request closed without merging**